src/lib.rs

```diff
@@ -606,15 +613,25 @@ impl Bump {
         unsafe {
             let footer = self.current_chunk_footer.get();
             let footer = footer.as_ref();
-            let ptr = footer.ptr.get().as_ptr() as usize;
+            let initial_ptr = footer.ptr.get().as_ptr() as usize;
             let start = footer.data.as_ptr() as usize;
-            debug_assert!(start <= ptr);
-            debug_assert!(ptr <= footer as *const _ as usize);
+            debug_assert!(start <= initial_ptr);
+            debug_assert!(initial_ptr <= footer as *const _ as usize);
 
-            let ptr = ptr.checked_sub(layout.size())?;
+            let ptr = initial_ptr.checked_sub(layout.size())?;
             let aligned_ptr = ptr & !(layout.align() - 1);
 
             if aligned_ptr >= start {
+                // We might be the very first allocation made in the entire bump,
+                // in which case we need to set end_of_first_allocation
+                // for the first chunk
+                if initial_ptr == footer as *const ChunkFooter as usize {
```

Comment:
I was hoping we would be able to avoid the check in the fast path and only do it in the slow path, but I realize now that we can't because of the very first allocation. Blech.